GitFyle.Core.Api/Services/Foundations/Contributions/ContributionService.Validations.cs

```diff
@@ -0,0 +1,175 @@
+                Message = $"Date is not recent. Expected a value between {startDate} and {endDate} but found {date}"
+            };
+        }
+
+        private async ValueTask<(bool IsNotRecent, DateTimeOffset StartDate, DateTimeOffset EndDate)>
+            IsDateNotRecentAsync(DateTimeOffset date)
+        {
+            int pastSeconds = 60;
+            int futureSeconds = 0;
+            DateTimeOffset currentDateTime = await this.dateTimeBroker.GetCurrentDateTimeOffsetAsync();
+
+            if (currentDateTime == default)
+            {
+                return (false, default, default);
+            }
+
+            TimeSpan timeDifference = currentDateTime.Subtract(date);
+            DateTimeOffset startDate = currentDateTime.AddSeconds(-pastSeconds);
+            DateTimeOffset endDate = currentDateTime.AddSeconds(futureSeconds);
+            bool isNotRecent = timeDifference.TotalSeconds is > 60 or < 0;
+
+            return (isNotRecent, startDate, endDate);
+        }
+
+        private static void ValidateContributionIsNotNull(Contribution contribution)
+        {
+            if (contribution is null)
+            {
+                throw new NullContributionException(message: "Contribution is null");
+            }
+        }
+
+        private static async ValueTask<dynamic> IsInvalidAsync(Guid id) => new
+        {
+            Condition = id == Guid.Empty,
+            Message = "Id is invalid"
+        };
+
+        private static async ValueTask<dynamic> IsInvalidAsync(string name) => new
+        {
+            Condition = String.IsNullOrWhiteSpace(name),
+            Message = "Text is required"
+        };
+
+        private static async ValueTask<dynamic> IsInvalidAsync(DateTimeOffset date) => new
+        {
+            Condition = date == default,
+            Message = "Date is invalid"
+        };
+
+        private static async ValueTask<dynamic> IsInvalidLengthAsync(string text, int maxLength) => new
+        {
+            Condition = await IsExceedingLengthAsync(text, maxLength),
+            Message = $"Text exceed max length of {maxLength} characters"
+        };
+
+        private static async ValueTask<bool> IsExceedingLengthAsync(string text, int maxLength) =>
+            (text ?? string.Empty).Length > maxLength;
+
+        private static async ValueTask<dynamic> IsNotSameAsync(
+           DateTimeOffset createdDate,
+           DateTimeOffset updatedDate,
+           string createdDateName) => new
+           {
+               Condition = createdDate != updatedDate,
+               Message = $"Date is not the same as {createdDateName}"
+           };
+
+        private static async ValueTask<dynamic> IsNotSameAsync(
+            string createBy,
+            string updatedBy,
+            string createdByName) => new
+            {
+                Condition = createBy != updatedBy,
+                Message = $"Text is not the same as {createdByName}"
+            };
```

Comment:
Update the method here:

```suggestion
        private static async ValueTask<dynamic> IsNotSameAsync(
            string first,
            string second,
            string secondName) => new
            {
                Condition = first != second,
                Message = $"Text is not the same as {secondName}"
            };
```